communicator/shared/shared.go

```diff
@@ -0,0 +1,26 @@
+package shared
+
+import (
+	"fmt"
+	"net"
+)
+
+// HostFormatter helps out with host formatting.
+// For instance, the host parameter differs when connecting to an IPv4 address vs. an IPv6 address.
+type HostFormatter interface {
+	Format(string) string
+}
+
+// HostFormatterImpl implements the HostFormatter interface.
+type HostFormatterImpl struct{}
+
+// Format formats the host/IP correctly, so we don't provide IPv6 address in an IPv4 format during node communication.
+func (h *HostFormatterImpl) Format(host string) string {
```

Comment:
Ok. I will change it tomorrow.
On Jun 7, 2016 10:36 PM, "Joseph Anthony Pasquale Holsten" <
notifications@github.com> wrote:

> In communicator/shared/shared.go
> <https://github.com/hashicorp/terraform/pull/6616#discussion_r66147721>:
>
> > +import (
> > +	"fmt"
> > +	"net"
> > +)
> > +
> > +// HostFormatter helps out with host formatting.
> > +// For instance, the host parameter differs when connecting to an IPv4 address vs. an IPv6 address.
> > +type HostFormatter interface {
> > +	Format(string) string
> > +}
> > +
> > +// HostFormatterImpl implements the HostFormatter interface.
> > +type HostFormatterImpl struct{}
> > +
> > +// Format formats the host/IP correctly, so we don't provide IPv6 address in an IPv4 format during node communication.
> > +func (h *HostFormatterImpl) Format(host string) string {
>
> I'd keep it simple until we actually need it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/hashicorp/terraform/pull/6616/files/52fb286766df7d25cfe8c18bf4b82e030b7faf06#r66147721>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAm9t4UPXRIBDwnzElU56kI-vt_3GWK7ks5qJdZhgaJpZM4IcN41>
> .
>